### PR TITLE
python: reformat and addition of format checks

### PR DESCRIPTION
**Pull request by trws**
Currently this applies the "black" formatter to our python code globally and adds two scripts, one that runs it to format all python code int the tree and another that checks whether formatting would change the code.  I'm planning to have the check script run in at least one of the travis targets, probably along with pylint.

The question now, is should I also add the clang-format formatting for our C code to this.  It's not materially harder, but before I do it, we need to pick a standard version to use (I suggest 6.0 or 7.0) and agree that the format file as defined in this PR (same as we had but disregarding support for 3.6, which is now ancient) is what we want code to look like or modify it such that it is.  Thoughts?

### CODE REVIEW

File: src/bindings/python/.pylintrc [hunk 1]
```diff
@@ -44,7 +44,7 @@ symbols=no
 # --enable=similarities". If you want to run only the classes checker, but have
 # no Warning level messages displayed, use"--disable=all --enable=classes
 # --disable=W"
-disable=missing-docstring, too-few-public-methods, too-many-arguments, star-args, arguments-differ, no-member, import-error, no-name-in-module, useless-object-inheritance, fixme
+disable=missing-docstring, too-few-public-methods, too-many-arguments, star-args, arguments-differ, no-member, import-error, no-name-in-module, useless-object-inheritance, fixme, C0330
```
Comment (SteVwonder) [inline src/bindings/python/.pylintrc]: Do you happen to know what the "human-readable" code for `C0330` is?  If so, can we use that instead?  Based on [this issue](https://github.com/PyCQA/pylint/issues/289), I believe it is `bad-continuation`.

Comment (trws) [reply src/bindings/python/.pylintrc]: I'll try that, good point.

**Comment by trws** [comment]
Definitely, I want this check to be part of travis, and I want this and pylint to run *first* rather than last as is being done now.  There's little more frustrating than compiling, checking and passing everything only to run the format check and fail at the end. 😖

**Comment by garlick** [comment]
> The question now, is should I also add the clang-format formatting for our C code to this.

I think we agreed that C should be a separate PR/issue.

In addition to this, we should probably also add a python section to RFC 7 to document our practices/expectations for people submitting PR's to framework projects.  Sounds like it can just reference the [the Black code style](https://github.com/ambv/black#the-black-code-style) instead of the specific details, if we are happy with that style.

**Comment by trws** [comment]
I have a locally rebased version of this waiting on #1795, having outstanding python PRs during a reformat would be unfortunate, should be very quick once the other merges.

**Comment by garlick** [comment]
@trws, I'm adding this to the 0.11.0 release milestone per your comments yesterday.

**Comment by trws** [comment]
Travis passed, but apparently codecov/project did not.  I'm not sure what to do about that given that this PR didn't actually add any new code (except the check script).  Perhaps it doesn't like that the code is now on more lines?

**Comment by SteVwonder** [comment]
Thanks @trws!

Other than the pylintrc comment above and some commit polishing, this LGTM!  I wouldn't worry about the codecov since it doesn't track python files.

Per the commit polishing, before merging, it would be good to squash some of the incremental commits and reword the commit titles to the form `topic: description`.

**Comment by trws** [comment]
Thanks @SteVwonder, the commits have been re-worked, and the bad-continuation option seems to work.  Assuming I didn't muck something else up and the rest of the travis run passes, we should be GTG.

**Comment by garlick** [comment]
I think that last commit could be squashed with the first one.

Making this a separate stage in travis looks really nice in the ci summary page!  @grondo could you give your ack on the travis  changes?

**Comment by grondo** [comment]
Oh yeah, Travis changes look great. Thanks, @trws!

**Comment by trws** [comment]
Good point, squashed and re-pushed.

**Comment by garlick** [comment]
Restarted a builder that failed the wreck epilog test.

Apologies @trws, this needs one more rebase then I'll press the button.

**Comment by garlick** [comment]
Thanks!

**Pull request merged**